include/fastdds/dds/xtypes/dynamic_types/DynamicTypeMember.hpp

```diff
@@ -0,0 +1,139 @@
+
+#include <fastdds/dds/core/ReturnCode.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/AnnotationDescriptor.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/MemberDescriptor.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/VerbatimTextDescriptor.hpp>
+#include <fastrtps/fastrtps_dll.h>
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+class DynamicTypeMember;
+
+template<>
+struct traits<DynamicTypeMember> : public object_traits<DynamicTypeMember>
+{
+    using base_type = DynamicTypeMember;
+};
+
+/*!
+ * Represents a "member" of a type. A "member" in this sense may be a member of an aggregated type, a constant within an
+ * enumeration, or some other type substructure.
+ */
+class RTPS_DllAPI DynamicTypeMember : public std::enable_shared_from_this<DynamicTypeMember>
+{
+public:
+
+    using _ref_type = typename traits<DynamicTypeMember>::ref_type;
+
+    /*!
+     * Provides a summary of the state of this type overwriting a provided object (see [standard] 7.5.2.6.2)
+     * @param[inout] descriptor @ref MemberDescriptor reference where copied the information.
+     * @return standard @ref ReturnCode_t
+     */
+    virtual ReturnCode_t get_descriptor(
+            traits<MemberDescriptor>::ref_type md) = 0;
+
+    /*!
+     * Returns the number of applied annotations to the member.
+     * @return Number of annotations.
+     */
+    virtual uint32_t get_annotation_count() = 0;
+
+    /*!
+     * Returns an applied annotation by index.
+     * @param[inout] descriptor @ref AnnotationDescriptor reference where copied the information.
+     * @param[in] idx Index.
+     * @return standard @ref ReturnCode_t
+     * @retval RETCODE_OK when the copy was successful.
+     * @retval RETCODE_BAD_PARAMETER when descriptor reference is nil or index is out-of-range.
+     */
+    virtual ReturnCode_t get_annotation(
+            traits<AnnotationDescriptor>::ref_type descriptor,
+            const uint32_t idx) = 0;
+
+    /*!
+     * Returns the number of applied verbatim text to the member.
+     * @return Number of verbatim texts.
+     */
+    virtual uint32_t get_verbatim_text_count() = 0;
+
+    /*!
+     * Returns an applied verbatim text by index.
+     * @param[inout] descriptor @ref VerbatimTextDescriptor reference where copied the information.
+     * @param[in] idx Index.
+     * @return standard @ref ReturnCode_t
+     * @retval RETCODE_OK when the copy was successful.
+     * @retval RETCODE_BAD_PARAMETER when descriptor reference is nil or index is out-of-range.
+     */
+    virtual ReturnCode_t get_verbatim_text(
+            traits<VerbatimTextDescriptor>::ref_type descriptor,
+            const uint32_t idx) = 0;
+
+    /**
+     * State comparison according with the [standard] sections \b 7.5.2.6.3
+     * @param[in] descriptor @DynamicTypeMember reference to compare to
+     * @return \b bool `true` on equality
+     */
+    virtual bool equals(
+            traits<DynamicTypeMember>::ref_type descriptor) = 0;
```

Comment:
The rest of the API has followed the API strictly, even the parameter names. This one should be

```suggestion
            traits<DynamicTypeMember>::ref_type other) = 0;
```

Reply:
Done in 1596a8a54